src/components/modes/student/StudentMode.js

```diff
@@ -4,30 +4,28 @@ import { connect } from 'react-redux';
 import StudentView from './StudentView';
 import { DEFAULT_VIEW, FEEDBACK_VIEW } from '../../../config/views';
 import { getAppInstanceResources } from '../../../actions';
-import Loader from '../../common/Loader';
 
 class StudentMode extends Component {
   static propTypes = {
     appInstanceId: PropTypes.string,
     view: PropTypes.string,
-    activity: PropTypes.number,
     dispatchGetAppInstanceResources: PropTypes.func.isRequired,
     userId: PropTypes.string,
   };
 
   static defaultProps = {
     view: 'normal',
     appInstanceId: null,
-    activity: 0,
     userId: null,
   };
 
-  constructor(props) {
-    super(props);
-    const { userId } = props;
+  state = { started: false };
+
+  componentDidMount() {
+    const { userId, dispatchGetAppInstanceResources } = this.props;
 
     // get the resources for this user
-    props.dispatchGetAppInstanceResources({ userId });
+    dispatchGetAppInstanceResources({ userId });
   }
 
   componentDidUpdate({ appInstanceId: prevAppInstanceId }) {
```

Comment:
Can you change this to:

```
  componentDidUpdate({ appInstanceId: prevAppInstanceId, userId: prevUserId }) {
    const {
      appInstanceId,
      dispatchGetAppInstanceResources,
      userId,
    } = this.props;
    // handle receiving the app instance id
    if (appInstanceId !== prevAppInstanceId || userId !== prevUserId) {
      dispatchGetAppInstanceResources({ userId });
    }
  }
```

To see if it fixes the bug we introduced, where the saved time was not loaded on refresh. Thanks!

Reply:
No it did not fix the bug. The saved time did not load on refresh